client/__tests__/RatingsAndReviews.test.js

```diff
@@ -1,14 +1,29 @@
 import React from 'react';
-import { render, screen, cleanup } from '@testing-library/react';
-// import { shallow } from 'enzyme';
-// import Adapter from 'enzyme-adapter-react-16';
-// * enzyme only supports up to react ^16.4.0-0, and the current dependency is 17.x
-// TODO: investigate how to replace enzyme functionality
+import { render, screen, cleanup, fireEvent } from '@testing-library/react';
 import RatingsAndReviews from '../src/components/RatingsAndReviews/RatingsAndReviews.jsx';
+import ProductDetailPage from '../src/components/ProductDetailPage.jsx';
 import { reviewsMock, reviewsMetaMock } from '../src/mockData/reviewsMock.js';
 
+var onChangeFileHandler = (event) => {
+  this.setState({
+    selectedImageFile: event.target.files[0],
+    fileLoaded: 0
+  });
+}
+
+var onClickUploadHandler = () => {
+  const data = new FormData();
+  data.append('file', this.state.selectedImageFile);
+  axios.post(`http://localhost:${port}/upload`, data, {
+  })
+    .then(response => {
+      console.log('successful upload: ', response);
+    });
+}
+
 beforeEach(() => {
-  render(<RatingsAndReviews productId={reviewsMock.product} reviews={reviewsMock.results} reviewsMeta={reviewsMetaMock}/>);
+  var ProductDetailPageMock = new ProductDetailPage;
+  render(<RatingsAndReviews top={ProductDetailPageMock} onChangeFileHandler={ProductDetailPageMock.onChangeFileHandler} onClickUploadHandler={ProductDetailPageMock.onClickUploadHandler} productId={reviewsMock.product} reviews={reviewsMock.results} reviewsMeta={reviewsMetaMock}/>);
```

Comment:
Stylistic meta nit: can you break out the attributes into new lines when it gets crazy long? Easier to read and grok what is happening all at once. 
```suggestion
  render(<RatingsAndReviews 
    top={ProductDetailPageMock} 
    onChangeFileHandler={ProductDetailPageMock.onChangeFileHandler} 
    onClickUploadHandler={ProductDetailPageMock.onClickUploadHandler} 
    productId={reviewsMock.product} 
    reviews={reviewsMock.results} 
    reviewsMeta={reviewsMetaMock}
  />);
```